### PR TITLE
Rename WP_Widget_{Type} PHP classes to WP_Widget_Media_{Type}

### LINKED ISSUE #60: Proposal: Rename WP_Widget_{Type} PHP classes to WP_Widget_Media_{Type}
We are using `media_` as the ID base prefix for the media widgets, but while the Image widget has `media_image` as its ID base the current class name is `WP_Widget_Image`. For consistency sake, it would make sense if we renamed it to `WP_Widget_Media_Image`.

### PR DESCRIPTION
**Pull request by westonruter**
Fixes #60.

### CODE REVIEW

File: wp-includes/widgets/class-wp-widget-media-video.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * @codeCoverageIgnore
  * @see WP_Widget
  */
-class WP_Widget_Video extends WP_Widget_Media {
+class WP_Widget_Media_Video extends WP_Widget_Media {
```
Comment (timmyc) [inline wp-includes/widgets/class-wp-widget-media-video.php:19]: in the doc block on line 3 it still is showing `WP_Widget_Video`

File: wp-includes/widgets/class-wp-widget-media-image.php [hunk 2]
```diff
@@ -14,7 +14,7 @@
  *
  * @see WP_Widget
  */
-class WP_Widget_Image extends WP_Widget_Media {
+class WP_Widget_Media_Image extends WP_Widget_Media {
```
Comment (timmyc) [inline wp-includes/widgets/class-wp-widget-media-image.php:17]: Likewise here the docblock on line 3 is showing the old class name of `WP_Widget_Image`

File: wp-includes/widgets/class-wp-widget-media-audio.php [hunk 3]
```diff
@@ -16,7 +16,7 @@
  * @codeCoverageIgnore
  * @see WP_Widget
  */
-class WP_Widget_Audio extends WP_Widget_Media {
+class WP_Widget_Media_Audio extends WP_Widget_Media {
```
Comment (timmyc) [inline wp-includes/widgets/class-wp-widget-media-audio.php:19]: Here too 😄  docblock has `WP_Widget_Audio`

**Review by timmyc — COMMENTED** [review]
Just some minor notes about the old class names appearing in other parts of the code still:

`WP_Widget_Image` is also still used in the following comments

```
tests/phpunit/test-class-wp-widget-image.php
3: * Unit tests covering WP_Widget_Image functionality.
14:class Test_WP_Widget_Image extends WP_UnitTestCase {

wp-admin/js/widgets/media-image-widget.js
10:	 * See WP_Widget_Image::enqueue_admin_scripts() for amending prototype from PHP exports.
20:	 * See WP_Widget_Image::enqueue_admin_scripts() for amending prototype from PHP exports.

wp-admin/js/widgets/media-widgets.js
185:		 * exported from PHP to JS such as is done in WP_Widget_Image::enqueue_admin_scripts().
198:		 * such as is done in WP_Widget_Image::enqueue_admin_scripts(). If not,
211:		 * such as is done in WP_Widget_Image::enqueue_admin_scripts().
533:		 * exported from PHP to JS such as is done in WP_Widget_Image::enqueue_admin_scripts().
```

**Comment by westonruter** [comment]
@timmyc good catches. Should be fixed in 65c6456.

**Review by timmyc — APPROVED** [review]
Looks good to me.

**Comment by westonruter** [comment]
@obenland any hesitations about this?

**Comment by obenland** [comment]
> @obenland any hesitations about this?

@westonruter I just thought of file names but you addressed that in 65c6456. It's not necessarily elegant but I guess that's the price you pay for backwards compatibility. :shipit:

**Review by obenland — APPROVED** [review]

**Pull request merged**